### DISCUSSION TITLE
Manual/class-PhysicsMaterial2D

### DISCUSSION THREAD

**Discussion by giscus** [Developer-Notes-Extension/Unity-Notes#691] [Notes]
# Manual/class-PhysicsMaterial2D

https://docs.unity3d.com/Manual/class-PhysicsMaterial2D.html

**Comment by JohnGroot** [comment]
As Unity uses Box2D for its 2D physics, the friction of 2D physics materials are solved via multiplication:
`float mixedFriction = sqrtf(b2Shape_GetFriction(shapeIdA) * b2Shape_GetFriction(shapeIdB));`
& bounciness is solved using the max of the two contacting bodies:
`float mixedRestitution = b2MaxFloat(b2Shape_GetRestitution(shapeIdA), b2Shape_GetRestitution(shapeIdB));`

Box2D docs on Friction for reference:
https://box2d.org/documentation/md_simulation.html#autotoc_md76
